src/plexil-adapter/OwAdapter.cpp

```diff
@@ -36,11 +37,54 @@ using std::endl;
 ///////////////////////////// Conveniences //////////////////////////////////
 
 // A prettier name for the "unknown" value.
-static Value const Unknown;
+//static Value const Unknown;
+const Value Unknown;
 
 // An empty argument vector.
 static vector<Value> const EmptyArgs;
 
+
+//////////////////////////// Command Handling //////////////////////////////
+
+static int CommandId = 0;
+
+static std::map<int, Command*> CommandRegistry;
+
+static void command_sent (Command* cmd)
+{
+  g_execInterface->handleCommandAck(cmd, COMMAND_SENT_TO_SYSTEM);
+  g_execInterface->notifyOfExternalEvent();
+}
+
+static void command_success (Command* cmd)
+{
+  g_execInterface->handleCommandAck(cmd, COMMAND_SUCCESS);
+  g_execInterface->notifyOfExternalEvent();
+}
+
+static void ack_command (Command* cmd, PLEXIL::CommandHandleValue handle)
+{
+  g_execInterface->handleCommandAck(cmd, handle);
+  g_execInterface->notifyOfExternalEvent();
+}
+
+static void command_status_callback (int id, bool success)
+{
+  Command* cmd;
+
+  try {
+    cmd = CommandRegistry.at(id);
+  }
+  catch (const std::out_of_range& oor) {
+    ROS_ERROR("command_status_callback: no command registered under id %d", id);
+    return;
+  }
+  // This ack is needed only when the command was registered.
+  if (success) ack_command (cmd, COMMAND_SUCCESS);
```

Comment:
I prefer newlines after ifs and elses, such that ack_command would be on the next line. However, I don't think this is required anywhere in our style guide.

Reply:
And I strongly prefer avoiding two extra lines of code if the statement fits into one line, and is otherwise readable in context.  Why _two_ extra lines?  Because I also strongly prefer that if/else clauses are bracketed even if they are only one line.  This _was_ a coding standard at my last company, because a common bug is inserting an extra indented line of code and thinking it goes with the 'if' of 'else' above it (ala Python).

Reply:
I don't have a strong preference. Have never seen this style required before.